### PR TITLE
Improvemets of PEB->CommandLine parsing algorithm

### PR DESCRIPTION
**Pull request by Forsari0**
- Try to find executable name _with_ extension instead of full launch _cmdline_ (**sh.exe** instead of **C:\sh.exe**)
- Try to find executable name _without_ extension instead of full launch _cmdline_ (**sh** instead of **C:\sh.exe**)

---
This change is [Reviewable](https://reviewable.io/reviews/x64dbg/x64dbg/1005)

### CODE REVIEW

**Comment by mrexodia** [comment]
What if you use the name of the executable in both the folder name and the
command line?

File: src/dbg/debugger.cpp [hunk 1]
```diff
@@ -1921,39 +1927,52 @@ bool dbglistprocesses(std::vector<PROCESSENTRY32>* infoList, std::vector<std::st
...
+                char* exeName = strrchr(pe32.szExeFile, '\\') ? strrchr(pe32.szExeFile, '\\') + 1 :
+                                strrchr(pe32.szExeFile, '/') ? strrchr(pe32.szExeFile, '/') + 1 : pe32.szExeFile;
+
+                char* exeNameInCmd = strstr(cmdline, exeName);
+                if(exeNameInCmd)
+                    cmdLineExeSize = (size_t)(((LPBYTE)exeNameInCmd - (LPBYTE)cmdline) + strlen(exeName));
+                else
+                {
+                    //try to locate basic name, without extension
+                    char* basicName = (char*)emalloc(strlen(exeName) + 1, "dbglistprocesses:basicName");;
```
Comment (mrexodia) [inline src/dbg/debugger.cpp]: This should be `Memory<char*> basicName(strlen(exeName) + 1);` which ensures the buffer is always freed on scope exit. Doesn't matter much though I can change it during the merge.

**Comment by Forsari0** [comment]
Okay, now it's correctly parse this occasion in both cases (_with_ and _without_ extension)

x64dbg_2016-08-25_18-20-34

**Comment by mrexodia** [comment]
Can I merge it or are you still working on it?

**Comment by Forsari0** [comment]
Based on my tests, it's pretty stable now, so you can safely merge it.

**Comment by mrexodia** [comment]
Thanks a lot!

**Pull request merged**